app/code/Magento/Msrp/etc/module.xml

```diff
@@ -9,6 +9,8 @@
     <module name="Magento_Msrp" >
         <sequence>
             <module name="Magento_Catalog"/>
+            <module name="Magento_MsrpConfigurableProduct"/>
+            <module name="Magento_MsrpGroupedProduct"/>
```

Comment:
Such dependency is incorrect. Those modules must depend on `Msrp` and not vise versa.

Reply:
@orlangur I don't know if you've understood the scenario of issue.

In > app/code/Magento/MsrpConfigurableProduct/etc/module.xml there is the dependency / sequence with <module name="Magento_Msrp"/>, so far, so good. 

In > app/code/Magento/MsrpGroupedProduct/etc/module.xml there is the dependency / sequence with <module name="Magento_Msrp"/>, so far, so good. 

But if you disable these modules MsrpConfigurableProduct and MsrpGroupedProduct, the module Magento_Msrp is dependent of MsrpConfigurableProduct or MsrpGroupedProduct.

With both modules MsrpConfigurableProduct and MsrpGroupedProduct disabled the application crashes.

The file app/code/Magento/Msrp/Pricing/MsrpPriceCalculator.php > public function __construct(array $msrpPriceCalculators) > it's supposed to receive as an argument MsrpConfigurableProduct or MsrpGroupedProduct

These are declared in:

app/code/Magento/MsrpConfigurableProduct/etc/di.xml

app/code/Magento/MsrpGroupedProduct/etc/di.xml

The way Magento_Msrp was built/developed is dependent of MsrpConfigurableProduct and MsrpGroupedProduct